src/main/java/org/opensearch/security/ssl/util/SSLRequestHelper.java

```diff
@@ -109,12 +109,16 @@ public static SSLInfo getSSLInfo(final Settings settings, final Path configPath,
             return null;
         }
 
-        final SslHandler sslhandler = (SslHandler) ((Netty4HttpChannel)request.getHttpChannel()).getNettyChannel().pipeline().get("ssl_http");
-        
+        final Netty4HttpChannel httpChannel = (Netty4HttpChannel)request.getHttpChannel();
+        SslHandler sslhandler = (SslHandler) httpChannel.getNettyChannel().pipeline().get("ssl_http");
+        if(sslhandler == null && httpChannel.inboundPipeline() != null) {
+            sslhandler = (SslHandler) httpChannel.inboundPipeline().get("ssl_http");
+        }
```

Comment:
For my curiosity, Can you help me understand why we are retrieving in sslHandler from an inboundPipeline() and why is this check required

Reply:
@DarshitChanpura if you look into previous code, it looked at channel's pipeline to find out the SSL handler. Now, with HTTP/2, the new pipeline is created (hence, the inbound channel pipeline is "lost") and the SSL handler is not accessible anymore. The solution (sort of): keep inbound pipeline reference so the SSL handler could be retrieved.